### PR TITLE
Update default param val attr

### PR DESCRIPTION
**Pull request by Rageking8**
## Summary

Changed up the first example and added a new example. Also rephrased the part that implied that C# does not have default parameters.

### CODE REVIEW

File: xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml [hunk 1]
```diff
@@ -64,16 +64,20 @@
...
   
- Note that the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> does not add support for default parameters to languages that do not support this feature.  For example, if you use the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> with a method written in C#, which does not support default parameters, you cannot use the default parameter when calling the method from C#.  You can use the default parameter only when calling the method in a language such as C++ that does support the feature.  
+ Note that the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> does not add support for default parameters to languages that do not support this feature.  For example, if you use the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> with a language that does not support default parameters, you cannot use the default parameter when calling the method in said language.  You can use the default parameter only when calling the method in a language such as C++ that does support the feature.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml:67]: I think all the .NET languages support it, so I'd suggest removing this paragraph.

File: xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml [hunk 2]
```diff
@@ -64,16 +64,20 @@
...
   
  The <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> is particularly useful to specify default parameters for methods of a COM interop interface.  
   
    
   
 ## Examples  
- The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#.  
+ The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#.  The <xref:System.Runtime.InteropServices.OptionalAttribute> attribute is used to allow calling of the method without any arguments.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml:74]:
```suggestion

The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#. The snippet also uses the <xref:System.Runtime.InteropServices.OptionalAttribute> attribute to enable the method to be called without any arguments.
```

File: xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml [hunk 3]
```diff
@@ -135,9 +139,9 @@
...
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml:142]:
```suggestion
 The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#. The <xref:System.Runtime.InteropServices.OptionalAttribute> attribute is also used to enable the method to be called without any arguments.
```

File: xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml [hunk 4]
```diff
@@ -64,16 +64,20 @@
...
    
   
 ## Examples  
- The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#.  
+ The following code example demonstrates how to apply the <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> attribute to a parameter of a method written in C#.  The <xref:System.Runtime.InteropServices.OptionalAttribute> attribute is used to allow calling of the method without any arguments.  
   
- :::code language="csharp" source="~/snippets/csharp/System.Runtime.InteropServices/DefaultParameterValueAttribute/Overview/sample.cs" id="Snippet1":::  
+ :::code language="csharp" source="~/snippets/csharp/System.Runtime.InteropServices/DefaultParameterValueAttribute/Overview/sample1.cs" id="Snippet1":::  
+  
+ Usage of <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> and <xref:System.Runtime.InteropServices.OptionalAttribute> does have some differences as compared to normal C# default parameters.
```
Comment (gewarren) [inline xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml:78]:
```suggestion
There are some behavior differences between [C# optional arguments](/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments) and using <xref:System.Runtime.InteropServices.DefaultParameterValueAttribute> and <xref:System.Runtime.InteropServices.OptionalAttribute>. The following code snippet demonstrates these differences.
```

Comment (Rageking8) [reply xml/System.Runtime.InteropServices/DefaultParameterValueAttribute.xml]: I replaced the `and` between the 2 attr xrefs with `with`. Hope that is better (current one has too many `and`s).

**Review by gewarren — APPROVED** [review]
Thanks @Rageking8, looks better.

**Pull request merged**